test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -669,6 +670,11 @@ public static void assertVersionSerializable(Version version, Streamable streama
                     equalTo(0));
             assertThat("Serialization failed with version [" + version + "] bytes should be equal for streamable [" + streamable + "]",
                     serialize(version, streamable), equalTo(orig));
+        } catch (UnsupportedOperationException ex) {
+            if (ex.equals("can't read named writeable from StreamInput")) {
```

Comment:
Surely if we hit an UnsupportedOperationException here it is a bug since all object should always be serialisable both before and after these changes?